pkg/catalog/catalog_test.go

```diff
@@ -8,15 +8,52 @@ import (
 )
 
 var _ = Describe("Catalog", func() {
+	var (
+		c       *catalog.Catalog
+		catName string
+	)
+
+	BeforeEach(func() {
+		c = catalog.New()
+		catName = "whiskers"
+	})
+
+	Context("Add", func() {
+		BeforeEach(func() {
+			profiles := []v1alpha1.ProfileDescription{{Name: "foo"}, {Name: "bar"}, {Name: "alsofoo"}}
+			c.Add(catName, profiles...)
+		})
+
+		It("adds the catalog name to each profile", func() {
+			Expect(c.Get("foo")).To(Equal(
+				v1alpha1.ProfileDescription{Name: "foo", Catalog: catName},
+			))
+		})
+	})
+
 	Context("Search", func() {
+		BeforeEach(func() {
+			profiles := []v1alpha1.ProfileDescription{{Name: "foo"}, {Name: "bar"}, {Name: "alsofoo"}}
+			c.Add(catName, profiles...)
+		})
+
 		It("returns matching profiles", func() {
-			c := catalog.New()
-			c.Add(v1alpha1.ProfileDescription{Name: "foo"})
-			c.Add(v1alpha1.ProfileDescription{Name: "bar"})
-			c.Add(v1alpha1.ProfileDescription{Name: "alsofoo"})
 			Expect(c.Search("foo")).To(ConsistOf(
-				v1alpha1.ProfileDescription{Name: "foo"},
-				v1alpha1.ProfileDescription{Name: "alsofoo"},
+				v1alpha1.ProfileDescription{Name: "foo", Catalog: catName},
+				v1alpha1.ProfileDescription{Name: "alsofoo", Catalog: catName},
+			))
+		})
+	})
+
+	Context("Get", func() {
+		BeforeEach(func() {
+			profiles := []v1alpha1.ProfileDescription{{Name: "foo"}, {Name: "bar"}, {Name: "alsofoo"}}
+			c.Add(catName, profiles...)
+		})
+
+		It("returns the requested profile", func() {
+			Expect(c.Get("foo")).To(Equal(
+				v1alpha1.ProfileDescription{Name: "foo", Catalog: catName},
```

Comment:
should this all be in 1 test with `By("")s`? It feels icky that the contexts are all cross-testing each other, e.g. if I  break `add` it also breaks `get` and `show`

Reply:
Yeh you are right, definitely not nice 👍